lib/lexer_mdx.mll

```diff
@@ -89,44 +65,35 @@ and cram_text section = parse
         newline lexbuf;
         `Section section :: cram_text (Some section) lexbuf }
   | "  " ([^'\n']* as first_line) eol
-      { let header = Some (Block.Header.Shell `Sh) in
+      { let start = Lexing.lexeme_start_p lexbuf in
+        newline lexbuf;
+        let header = "sh" in
         let requires_empty_line, contents = cram_block lexbuf in
         let contents = first_line :: contents in
-        let labels = [] in
-        let legacy_labels = false in
-        let loc = Location.curr lexbuf in
-        List.iter (fun _ -> newline lexbuf) contents;
+        let label_cmt = Some "" in
+        let legacy_labels = "" in
+        let end_ = Lexing.lexeme_start_p lexbuf in
         let rest = cram_text section lexbuf in
+        let loc = loc ~start ~end_ in
         let block =
-          match
-            Block.mk ~loc ~section ~header ~contents ~labels
+            Block.Raw.make ~loc ~section ~header ~contents ~label_cmt
               ~legacy_labels ~errors:[]
-          with
-          | Ok block -> block
-          | Error (`Msg msg) -> failwith msg
         in
         `Block block
         :: (if requires_empty_line then `Text "" :: rest else rest) }
   | "<-- non-deterministic" ws* ([^'\n']* as choice) eol
-      { let header = Some (Block.Header.Shell `Sh) in
-        let requires_empty_line, contents = cram_block lexbuf in
-        let labels =
-          match Label.interpret "non-deterministic" (Some (Eq, choice)) with
-          | Ok label -> [label]
-          | Error (`Msg msg) -> failwith msg
-        in
-        let legacy_labels = false in
+      { let start = Lexing.lexeme_start_p lexbuf in
         newline lexbuf;
-        let loc = Location.curr lexbuf in
-        List.iter (fun _ -> newline lexbuf) contents;
+        let header = "sh" in
+        let requires_empty_line, contents = cram_block lexbuf in
+        let label_cmt = Some (Printf.sprintf "non-deterministic=%s" choice) in
```

Comment:
It's not a big thing, but having helper functions to construct known labels could be useful

Reply:
Here it's specifically to deal with a special in the cram syntax which is deprecated so I went for the simplest and most straightforward implementation.